### PR TITLE
checker: disallow deferencing a `nil` pointer

### PR DESCRIPTION
**Pull request by Delta456**
### <samp>🤖 Generated by Copilot at 262c192</samp>

Add a checker error and a test case for dereferencing a nil pointer. This improves the compiler's ability to detect unsafe code that could cause a segmentation fault.

### <samp>🤖 Generated by Copilot at 262c192</samp>

*  Add a check for dereferencing a nil pointer in the checker module ([link](https://github.com/vlang/v/pull/18038/files?diff=unified&w=0#diff-22c1eb3704a15ee5526dec7b8cfe3f172f040b0aabfc682f2d76d55d174cac46R3879-R3889))
*  Add a test case for the nil pointer dereference check ([link](https://github.com/vlang/v/pull/18038/files?diff=unified&w=0#diff-d2907434d3970e418ae2497200d165ed8f3c9f9ade46e110553b338c58758984R1-R5), [link](https://github.com/vlang/v/pull/18038/files?diff=unified&w=0#diff-9ef21ca233e113f27854d9747a7fde170df5f211695e96d1abd4f315d3147f64R1-R3))

Fixes https://github.com/vlang/v/issues/18034

### CODE REVIEW

File: vlib/v/checker/checker.v [hunk 1]
```diff
@@ -3876,6 +3876,17 @@ fn (mut c Checker) prefix_expr(mut node ast.PrefixExpr) ast.Type {
 		if right_type.is_voidptr() {
 			c.error('cannot dereference to void', node.pos)
 		}
+		if mut node.right is ast.Ident {
+			if mut var := node.right.scope.find_var('${node.right.name}') {
+				// Workaround for unused variable warning
+				_ = var
```
Comment (Delta456) [inline vlib/v/checker/checker.v:3882]: I hope if this workaround wasn't needed but it is:

```
vlib/v/checker/checker.v:3880:11: warning: unused variable: `var`
 3878 |         }
 3879 |         if mut node.right is ast.Ident {
 3880 |             if mut var := node.right.scope.find_var('${node.right.name}') {
      |                    ~~~
 3881 | 

```

Comment (medvednikov) [reply vlib/v/checker/checker.v:3882]: That's a bug in the checker. It should mark `var` as used here. Can you create a small github issue for this bug please?

Comment (joe-conigliaro) [reply vlib/v/checker/checker.v:3882]: I dont know why this has mut `if mut var := node....` My only guess is it used to modify var at some stage, or the code was copied from somewhere else?

I guess mut variable that is never mutated could be a warning or error.
